Directory.Build.props

```diff
@@ -58,15 +54,11 @@
     </PropertyGroup>
 
     <ItemGroup>
-        <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="5.0.3" />
-        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="3.11.0" />
-        <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.2" />
-        <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="16.10.56" />
+        <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="6.0.0" />
+        <PackageReference Include="Microsoft.CodeAnalysis.CSharp" Version="4.2.0" />
+        <PackageReference Include="Microsoft.CodeAnalysis.Analyzers" Version="3.3.3" />
+        <PackageReference Include="Microsoft.VisualStudio.Threading.Analyzers" Version="17.2.32" />
         <PackageReference Include="StyleCop.Analyzers" Version="1.2.0-beta.435" />
-        <PackageReference Condition=" '$(IncludeCompatabilityAnalyzer)' == 'True' " Include="Microsoft.DotNet.Analyzers.Compatibility" Version="0.2.12-alpha">
```

Comment:
NOTE: Removed this because it's been replaced with the built-in analyzers of .NET 5 and later versions,  according to https://www.nuget.org/packages/Microsoft.DotNet.Analyzers.Compatibility/0.2.12-alpha